platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -492,35 +432,35 @@ public long getLastLeafPath() {
     }
 
     /**
-     * Pauses merging of all data file collections used by this data source. It may not stop merging
-     * immediately, but as soon as merging process needs to update data source state, which is
+     * Pauses compaction of all data file collections used by this data source. It may not stop compaction
+     * immediately, but as soon as compaction process needs to update data source state, which is
      * critical for snapshots (e.g. update an index), it will be stopped until {@link
-     * #resumeMerging()}} is called.
+     * #resumeCompaction()}} is called.
      */
-    void pauseMerging() throws IOException {
+    void pauseCompaction() throws IOException {
```

Comment:
This class creates a compactor for each instance.  Would it make sense to have pause and resume in that class, or is there intention to change that in the future to share a compactor among more than one instance of this class (and/or other classes).

Reply:
There is no intention to share the instances of `DataFileCompactor` or `MerkleDbCompactionCoordinator` between instances of `VirtualMap`-s. These classes have states that are specific to a particular VM and that state is bound to VM's lifecycle. The only thing VMs really share in the context of compaction is the thread pool and the only reason to share it is to limit resource consumption, as the compaction has lesser priority than transaction processing.  Other than that, the compaction in different VM is totally independent.

Reply:
So to the question.  Does it make sense to move the pause and resume into the `MerkleDbCompactionCoordinator` rather than have each item paused/resumed in this class?

Reply:
Yes, `DataFileCompactor` can be extracted from classes that implement `Compactible`. I'll try it out